akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/BehaviorTestKitImpl.scala

```diff
@@ -132,7 +133,36 @@ private[akka] final class BehaviorTestKitImpl[T](_path: ActorPath, _initialBehav
     try {
       context.setCurrentActorThread()
       try {
-        currentUncanonical = Behavior.interpretMessage(current, context, message)
+        //we need this to handle message adapters related messages
+        val intercepted0 = Behaviors.intercept{ () =>
+          new BehaviorInterceptor[Any, T]() {
```

Comment:
I think we could define this as a concrete class rather than anonymous inner and re-use the same `BehaviorInterceptor` instance for all calls.